app/models/voucher.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Voucher < ApplicationRecord
+  validates :code, presence: true
+  validates :expiration_date, presence: true
+  validates :points_required, presence: true, numericality: { only_integer: true }
+  validates :reward, presence: true
+
+  has_many :user_vouchers, dependent: :destroy
+end
```

Comment:
You can keep it, but for the future - you called this "feature/authentication" and those vouchers are not related to that at all. It will be easier for you to work with and for reviewers to go through your changes if you focus on a single thing that given PR adds - those vouchers could be added in other PR. That way it would be less code to review so you would have a quicker feedback loop meaning that you will be able to merge this faster.

Reply:
For someone who's aiming to be the "single-responsibility-_everything_" zealot that's quite a mess. Thanks for bringing my attention to this.